src/Frontend/Modules/Profiles/Actions/ChangeEmail.php

```diff
@@ -38,94 +37,79 @@ class ChangeEmail extends FrontendBaseBlock
 
     public function execute(): void
     {
-        // profile logged in
-        if (FrontendProfilesAuthentication::isLoggedIn()) {
-            parent::execute();
-            $this->getData();
-            $this->loadTemplate();
-            $this->buildForm();
-            $this->validateForm();
-            $this->parse();
-        } else {
-            // profile not logged in
-            $this->redirect(
-                FrontendNavigation::getUrlForBlock(
-                    'Profiles',
-                    'Login'
-                ) . '?queryString=' . FrontendNavigation::getUrlForBlock('Profiles', 'ChangeEmail'),
-                307
-            );
+        if (!FrontendProfilesAuthentication::isLoggedIn()) {
+            throw new InsufficientAuthenticationException('You need to log in to change your email');
```

Comment:
If you are not logged in, you can't know if the user has insufficient rights, so a Forbidden would be more logical

Reply:
see previous comment, this is how symfony handles it